dart/lib/src/hub.dart

```diff
@@ -0,0 +1,379 @@
+import 'dart:async';
+import 'dart:collection';
+
+import 'client.dart';
+import 'protocol.dart';
+import 'scope.dart';
+import 'sentry_options.dart';
+
+typedef ScopeCallback = void Function(Scope);
+
+/// SDK API contract which combines a client and scope management
+class Hub implements HubInterface {
+  static SentryClient _getClient({SentryOptions fromOptions}) {
+    return SentryClient(
+      dsn: fromOptions.dsn,
+      environmentAttributes: fromOptions.environmentAttributes,
+      compressPayload: fromOptions.compressPayload,
+      httpClient: fromOptions.httpClient,
+      clock: fromOptions.clock,
+      uuidGenerator: fromOptions.uuidGenerator,
+    );
+  }
+
+  final ListQueue<_StackItem> _stack;
+
+  final SentryOptions _options;
+
+  factory Hub(SentryOptions options) {
+    _validateOptions(options);
+
+    return Hub._(options);
+  }
+
+  Hub._(SentryOptions options)
+      : _options = options,
+        _stack = ListQueue() {
+    _stack.add(_StackItem(_getClient(fromOptions: options), Scope(_options)));
+    _isEnabled = true;
+  }
+
+  static void _validateOptions(SentryOptions options) {
+    if (options == null) {
+      throw ArgumentError.notNull('options');
```

Comment:
mm I remember asking in other PR about the `assert(condition)` and you said it was the only way, TIL `ArgumentError.notNull`, should we only use this one them?

Reply:
Good point ! I'm used to code mostly in Flutter where `assert` are executed only in debug mode, but as it's the Dart package so, you right, assert could be used here ( before NNBD ).

Reply:
Well, finally...
As the package can be used in multiple context ( like flutter or browser ), the asserts will not be executed, and it can leads to NullError. While NNBD is not landed this is the best option.

Reply:
mm I'm lost here now, which one is the best option? `assert` or `ArgumentError.notNull`?

Reply:
sorry, for now the best option is to test null value and throw `ArgumentError.notNull`.